### PR TITLE
Added default file extensions if missing

### PR DESCRIPTION
**Pull request by benmpeterson**
created constants for file extensions

Using io.FileNameUtils to check extension type
inline submissions missing extension defualt .html
uploaded files missing extension default to .docx

moved fileName creation to inside stage 1 as it is the only place used.

### CODE REVIEW

File: content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java [hunk 1]
```diff
@@ -659,23 +664,32 @@ public void processUnsubmitted(Calendar cal) {
...
+					if (INLINE_SUBMISSION.equals(fileName)) {
+						fileName += HTML_EXTENSION;
+						// Else add .docx
+					} else {
+						fileName += DOCX_EXTENSION;
+					}
+				}
+				// If fileName is empty set default
+				if (StringUtils.isEmpty(fileName)) {
+					fileName = "submission_" + item.getUserId() + "_" + item.getSiteId();
```
Comment (baholladay) [inline content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: default filename has no extension

File: content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java [hunk 2]
```diff
@@ -112,6 +113,10 @@
 	private static final String HEADER_AUTH = "Authorization";
 	private static final String HEADER_CONTENT = "Content-Type";
 	private static final String HEADER_DISP = "Content-Disposition";
+	
+	private static final String INLINE_SUBMISSION = "Inline Submission";
```
Comment (baholladay) [inline content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: remove

File: content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java [hunk 3]
```diff
@@ -659,23 +664,32 @@ public void processUnsubmitted(Calendar cal) {
...
-			}
 			// EXTERNAL ID DOES NOT EXIST, CREATE SUBMISSION AND UPLOAD CONTENTS TO TCA
 			// (STAGE 1)
 			if (StringUtils.isEmpty(item.getExternalId())) {
+				// Get filename of submission
+				String fileName = resource.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME);
+				// If file name is missing extension add to fulfill TCA header requirement
+				if (FilenameUtils.getExtension(fileName).isEmpty()) {
+					// Add .html for in line submissions
+					if (INLINE_SUBMISSION.equals(fileName)) {
```
Comment (baholladay) [inline content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: This isn't a good check and makes assumptions on the title and the language. There is an "inline submission" flag property associated with the file itself:

ResourcePropertiesEdit inlineProps = contentHostingService.newResourceProperties();
        inlineProps.addProperty(ResourceProperties.PROP_DISPLAY_NAME, rb.getString("submission.inline"));
        inlineProps.addProperty(ResourceProperties.PROP_DESCRIPTION, resourceId);
        inlineProps.addProperty(AssignmentConstants.PROP_INLINE_SUBMISSION, "true");

File: content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java [hunk 4]
```diff
@@ -659,23 +664,32 @@ public void processUnsubmitted(Calendar cal) {
...
-				fileName = "submission_" + item.getUserId() + "_" + item.getSiteId();
-				log.info("Using Default Filename " + fileName);
-			}
 			// EXTERNAL ID DOES NOT EXIST, CREATE SUBMISSION AND UPLOAD CONTENTS TO TCA
 			// (STAGE 1)
 			if (StringUtils.isEmpty(item.getExternalId())) {
+				// Get filename of submission
+				String fileName = resource.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME);
+				// If file name is missing extension add to fulfill TCA header requirement
+				if (FilenameUtils.getExtension(fileName).isEmpty()) {
```
Comment (baholladay) [inline content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: does "getExtension" work with a null parameter?

File: content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java [hunk 5]
```diff
@@ -659,23 +664,32 @@ public void processUnsubmitted(Calendar cal) {
...
+					// Add .html for in line submissions
+					if (INLINE_SUBMISSION.equals(fileName)) {
+						fileName += HTML_EXTENSION;
```
Comment (baholladay) [inline content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: why is docx the default? Is there a more generic option?

Comment (benmpeterson) [reply content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: perhaps .txt would be better here?

Comment (baholladay) [reply content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: let's leave this for now, we may remove the default completely if ends up not being required

Comment (baholladay) [reply content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: I talked w/Chris... do not set an extension if you don't know it. In the "inline submission" case, we know it is "html", so keep that... remove the default docx completely

Comment (benmpeterson) [reply content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: okay, what about for default filename?

Comment (baholladay) [reply content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: Leave a default filename, just w/o extension

File: content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java [hunk 6]
```diff
@@ -659,23 +662,27 @@ public void processUnsubmitted(Calendar cal) {
...
+				String fileName = resource.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME);						
+				// If fileName is empty set default
+				if (StringUtils.isEmpty(fileName)) {
+					fileName = "submission_" + item.getUserId() + "_" + item.getSiteId();
+					log.info("Using Default Filename " + fileName);
+				}				
+				// Add .html for inline submissions
+				ResourceProperties resourceProperties = resource.getProperties();
+				if ("true".equals(resourceProperties.getProperty(AssignmentConstants.PROP_INLINE_SUBMISSION))) {
+					fileName += HTML_EXTENSION;
```
Comment (baholladay) [inline content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: only do this if there is no file extension already

File: content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java [hunk 7]
```diff
@@ -659,23 +662,27 @@ public void processUnsubmitted(Calendar cal) {
...
 			if (StringUtils.isEmpty(item.getExternalId())) {
+				// Get filename of submission						
+				String fileName = resource.getProperties().getProperty(ResourceProperties.PROP_DISPLAY_NAME);						
+				// If fileName is empty set default
+				if (StringUtils.isEmpty(fileName)) {
+					fileName = "submission_" + item.getUserId() + "_" + item.getSiteId();
+					log.info("Using Default Filename " + fileName);
+				}				
+				// Add .html for inline submissions
+				ResourceProperties resourceProperties = resource.getProperties();
```
Comment (baholladay) [inline content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: why do you store resourceProperties as a variable here but not on line #669?

**Pull request merged**